doc/nrf/security/tfm.rst

```diff
@@ -254,38 +266,447 @@ We will decrease the size of the (optional) ``mcuboot_pad`` partition and thus t
+
+With |NCS|, the following commands can be used to generate the reports:
+
+  .. code-block:: console
+
+    west build -p -b nrf9151dk/nrf9151/ns samples/tfm/tfm_hello_world
+    cd build/tfm_hello_world
+    ninja tfm_ram_report
+    ninja tfm_rom_report
+
+Reports can be used to analyze the memory usage of the different TF-M partitions and how changing the Kconfig options affects the memory usage.
+For more information about the ``tfm_ram_report`` and ``tfm_rom_report`` targets, refer to the :ref:`tfm_build_system` documentation.
+
+TF-M Partitions
+===============
+
+CONFIG_TFM_PARTITION_PLATFORM
+-----------------------------
+
+Platform partition provides platform-specific services to other TF-M partitions and to non-secure application.
+
+For the non-secure application, setting the :kconfig:option:`CONFIG_TFM_ALLOW_NON_SECURE_FAULT_HANDLING` Kconfig option enables more detailed error descriptions of faults from the non-secure application with the Zephyr fault handler.
+
+Platform partition also exposes the following |NCS| specific APIs for the non-secure application:
+
+  .. code-block:: c
+
+    /* Search for the fw_info structure in firmware image located at address. */
+    int tfm_platform_firmware_info(uint32_t fw_address, struct fw_info *info);
+
+    /* Check if S0 is the active B1 slot. */
+    int tfm_platform_s0_active(uint32_t s0_address, uint32_t s1_address, bool *s0_active);
+
+See :ref:`lib_tfm_ioctl_api`: for more information about what APIs are available for the non-secure application.
+
+For more information about the general features of the TF-M Platform partition, see `TF-M Platform`_.
+
+CONFIG_TFM_PARTITION_INTERNAL_TRUSTED_STORAGE
+---------------------------------------------
+
+Internal Trusted Storage (ITS) partition implements the PSA Internal Trusted Storage APIs (`PSA Certified Secure Storage API 1.0`_) to achieve confidentiality, authenticity and encryption in rest (optional).
+ITS is meant to be used by the other TF-M partitions. It must not be accessed directly by the non-secure application, instead the :ref:`tfm_partition_ps` should be used.
+
+For more information about the general features of the TF-M ITS service, see `TF-M ITS`_.
+
+.. _tfm_encrypted_its:
+
+Encrypted ITS
+~~~~~~~~~~~~~
+
+To enable TF-M ITS encryption, use the Kconfig option :kconfig:option:`CONFIG_TFM_ITS_ENCRYPTED`.
+The ITS encryption is transparent to the user as long as the Master Key Encryption Key (MKEK) is populated before use.
+
+On Nordic Semiconductor devices, the hardware-accelerated AEAD scheme ChaChaPoly1305 is used with a 256 bits key.
+This key is derived with a key derivation function (KDF) based on NIST SP 800-108 CMAC.
+The input key of the KDF is the MKEK, a symmetric key stored in the Key Management Unit (KMU) of Nordic Semiconductor devices.
+The MKEK is protected by the KMU peripheral and its key material cannot be read by software. It can only be used by reference.
```

Comment:
```suggestion
The MKEK is protected by the KMU peripheral and its key material cannot be read by the software.
It can only be used by reference.
```